internal/pkg/api/cluster.go

```diff
@@ -126,6 +126,25 @@ func WithControlPlaneLabel(key string, val string) ClusterFiller {
 	}
 }
 
+// WithControlPlaneAPIServerExtraArgs adds the APIServerExtraArgs to the cluster spec.
+func WithControlPlaneAPIServerExtraArgs(key string, val string) ClusterFiller {
+	return func(c *anywherev1.Cluster) {
+		if c.Spec.ControlPlaneConfiguration.APIServerExtraArgs == nil {
+			c.Spec.ControlPlaneConfiguration.APIServerExtraArgs = map[string]string{}
+		}
+		c.Spec.ControlPlaneConfiguration.APIServerExtraArgs[key] = val
```

Comment:
Did we figure out what happens if we specify both this and pod iam config?

Reply:
`service-account-issuer` flag in `spec.controlPlaneConfiguration.apiServerExtraArgs` overrides the one in `spec.podIamConfig` based on the order in which we are appending the values. I am yet to check if we can append those values as a comma-separated string so that both can be enabled

Reply:
Added support to enable both config. PodIamConfig will be appended after the existing controlPlaneConfiguration `service-account-issuer` as a comma-separated string

Reply:
So `podIamConfig` will superseed the service-account-issuer flag? 

That detail needs to be included in the documentation for both `podIamConfig` and the `apiServerExtraArgs`. I'm open to docs being done in a separate PR, but we can't forget to add it in _both places_.

Reply:
No, `podIamConfig` won't supersede the service-account-issuer flag. If the `service-account-issuer` is already configured with `apiServerExtraArgs`, then `podIamConfig` will just append to it, making sure that both issuer urls are accepted for the flag

I was planning to create a separate PR for the docs. I will include this detail there